homeassistant/components/notify/slack.py

```diff
@@ -22,6 +23,8 @@
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_API_KEY): cv.string,
     vol.Required(CONF_CHANNEL): cv.string,
+    vol.Optional(CONF_USERNAME, default=None): cv.string,
+    vol.Optional(CONF_ICON, default=None): cv.string,
```

Comment:
`as_user` should be an additional variable. Also, consider to use `vol.Exclusive`/`vol.Inclusive` if it helps the user.